src/server/main_service.cc

```diff
@@ -695,6 +695,8 @@ void Service::Init(util::AcceptServer* acceptor, std::vector<facade::Listener*>
   config_registry.RegisterMutable("masterauth");
   config_registry.RegisterMutable("tcp_keepalive");
   config_registry.RegisterMutable("replica_partial_sync");
+  config_registry.Register("max_eviction_per_heartbeat");
```

Comment:
you register the flags to config registry, this allows you to set the flag to new value, but because you use the flag only on db_slice initialisation this has no impact, if you want to be able to change the value using config set you will need to call GetFlag(FLAGS_max_eviction_per_heartbeat) inside the logic that uses this value to get the most updated value

Reply:
He registered them as read-only

Reply:
initially I didnt intend to let user change it while running, but it could be a convenient thing to support. I can do some change to register it as a mutable one...

Reply:
just changed them to mutable and update these two values before each eviction. Please help to check if this is an appropriate way. Thanks!